config.m4

```diff
@@ -406,7 +406,18 @@ if test "$PHP_MONGODB_SASL" != "no"; then
   ], [
     -L$MONGODB_SASL_DIR/$PHP_LIBDIR
   ])
+
+  PHP_CHECK_LIBRARY(sasl2, sasl_client_done,
+  [
+    AC_SUBST(MONGOC_HAVE_SASL_CLIENT_DONE, 1)
+  ], [
+    AC_SUBST(MONGOC_HAVE_SASL_CLIENT_DONE, 0)
+  ])
+else
+  AC_SUBST(MONGOC_ENABLE_SASL, 0)
```

Comment:
I realized that `MONGOC_ENABLE_SASL` was also missing when we configured without SASL (same issue that you brought up with `MONGOC_HAVE_SASL_CLIENT_DONE`). Relevant libmongoc changes are https://github.com/mongodb/mongo-c-driver/commit/c3c14f431864c82f670581c0e380cef9047cfb96.